### ISSUE TITLE
Parameterizing scale_tril by scale and corr_tril

### ISSUE THREAD

**Issue by fehiepsi** [pyro-ppl/pyro#2820]
In #2812, we parameterize `scale_tril` by a global scale `scale` and `scale_tril`: `scale_tril = scale[..., None] * scale_tril`, which helps in learning the global `scale`. Alternatively, we might consider using the pair `scale` and `corr_tril` which do not suffer from 2 sources of over-parametrize:
+ unconstrained -> scale_tril: unused triu parameters
+ scale * scale_tril: doubling parameters of the diagonal parts

We can use this pair (`scale` and `corr_tril`) in AutoMVN distribution too. But `CorrCholeskyTransform` can be expensive and numerically unstable. So we need to do some benchmarks first. This issue is created to keep track of the above observations.

**Comment by fritzo** [comment]
@fehiepsi I agree we should switch the `AutoMultivariateNormal` to use a `(scale, scale_tril)` overparametrizaion.

Is there a particular reason you'd like to avoid overparametrizing?  While bijections are required in MCMC, I'm unclear why we might want to restrict to bijections for learnable parameters (hence the original distinction between `biject_to()` and `trasform_to()`).  My intuition is that elementwise optimizers like Adam can struggle with correlated changes, hence overparametrization can help by giving those optimizers multiple alternative directions of descent.  I believe @martinjankowiak has similar intuition from his [`OMTMultivariateNormal` distribution](https://docs.pyro.ai/en/stable/distributions.html#omtmultivariatenormal) and [OMT paper](http://proceedings.mlr.press/v89/jankowiak19a/jankowiak19a.pdf).
